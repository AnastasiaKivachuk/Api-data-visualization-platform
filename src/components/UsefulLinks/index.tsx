import React from "react";
import styled from "@emotion/styled";
import Link from "next/link";

import Typography from "@mui/material/Typography";
import Grid from "@mui/material/Grid";
import ArticleIcon from "@mui/icons-material/Article";
import YouTubeIcon from "@mui/icons-material/YouTube";
import SchoolIcon from "@mui/icons-material/School";

import {LaunchLinksInterface} from "../../interfaces/lauch.interface";

type Props = {
    launchLinks: LaunchLinksInterface,

}

const icons = {article_link: <ArticleIcon/>, video_link: <YouTubeIcon/>, wikipedia: <SchoolIcon/>}

const UsefulLinks = ({launchLinks}: Props): JSX.Element => {

    const Wrap = styled.div`
      margin: 20px auto;
    `

    // @ts-ignore
    const getLinks = (launchLinks) = Object.keys(launchLinks).map(item => ({icon: icons[item], url: launchLinks[item]})).filter(item => item?.icon);

    return (
        <Wrap>
            <Typography variant='h3' textAlign='center'>Usefully Links:</Typography>
            <Grid gap='20px' container justifyContent={'center'}>
                {getLinks.map(item => item.url &&
                    <Link target={'_blank'} key={item.url} href={item.url}><a>{item.icon}</a></Link>)}
            </Grid>
        </Wrap>
    )
}

export default UsefulLinks
