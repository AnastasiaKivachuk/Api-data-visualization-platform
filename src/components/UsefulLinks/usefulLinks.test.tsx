import {shallow} from "enzyme";
import React from "react";

import ArticleIcon from "@mui/icons-material/Article";
import YouTubeIcon from "@mui/icons-material/YouTube";
import SchoolIcon from "@mui/icons-material/School";

import UsefulLinks from "./index";

const MockLinks = {
    article_link: "http://spacenews.com/38959spacex-delivers-thaicom-6-satellite-to-orbit/",
    flickr_images: ['https://farm9.staticflickr.com/8617/16789019815_f99a165dc5_o.jpg', 'https://farm8.staticflickr.com/7619/16763151866_35a0a4d8e1_o.jpg', 'https://farm9.staticflickr.com/8569/16169086873_4d8829832e_o.png'],
    mission_patch_small: "https://images2.imgbox.com/d8/6d/fnqIBEJh_o.png",
    video_link: "https://www.youtube.com/watch?v=AnSNRzMEmCU",
    wikipedia: "https://en.wikipedia.org/wiki/Thaicom_6"
}

const component = shallow(<UsefulLinks launchLinks={MockLinks}/>);

describe("UsefulLinks component", () => {
        it("should render Menu component", () => {
            expect(component).toMatchSnapshot();
        })
        it("check all elements", () => {
            expect(component.find('a').length).toBe(3)
        })

        it("check  getLinks links method", () => {
            const result = [
                "http://spacenews.com/38959spacex-delivers-thaicom-6-satellite-to-orbit/", "https://www.youtube.com/watch?v=AnSNRzMEmCU", "https://en.wikipedia.org/wiki/Thaicom_6"
            ]

            component.find('ForwardRef').map((item, index) =>
                expect(item.prop('href')).toBe(result[index]))

        })

    }
)