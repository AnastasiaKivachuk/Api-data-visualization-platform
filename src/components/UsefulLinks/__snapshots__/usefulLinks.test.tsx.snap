// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsefulLinks component should render Menu component 1`] = `
<Styled(div)>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h3"
  >
    Usefully Links:
  </ForwardRef(Typography)>
  <ForwardRef(Grid)
    container={true}
    gap="20px"
    justifyContent="center"
  >
    <ForwardRef
      href="http://spacenews.com/38959spacex-delivers-thaicom-6-satellite-to-orbit/"
      key="http://spacenews.com/38959spacex-delivers-thaicom-6-satellite-to-orbit/"
      target="_blank"
    >
      <a>
        <Memo(ForwardRef(ArticleIcon)) />
      </a>
    </ForwardRef>
    <ForwardRef
      href="https://www.youtube.com/watch?v=AnSNRzMEmCU"
      key="https://www.youtube.com/watch?v=AnSNRzMEmCU"
      target="_blank"
    >
      <a>
        <Memo(ForwardRef(YouTubeIcon)) />
      </a>
    </ForwardRef>
    <ForwardRef
      href="https://en.wikipedia.org/wiki/Thaicom_6"
      key="https://en.wikipedia.org/wiki/Thaicom_6"
      target="_blank"
    >
      <a>
        <Memo(ForwardRef(SchoolIcon)) />
      </a>
    </ForwardRef>
  </ForwardRef(Grid)>
</Styled(div)>
`;
